vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -474,6 +474,158 @@ def query_metrics_wrapper(self, query_specs):
+                        hostname=self._hostname,
+                    )
+
+            for cluster in new_performance_metrics:
+                for entity_type in cluster:
+                    if len(entity_type.value) > 0:
+                        resource_type = entity_type.value[0].metricId.dynamicProperty[0][0]
+                        for given_metric in entity_type.value:
+                            self.log.debug(
+                                "Processing metric with type:%s",
+                                type(given_metric),
+                            )
+                            latest_value = given_metric.values.split(',')[-1]
+                            if latest_value != 'None':
+                                if given_metric.metricId.label in VSAN_PERCENT_METRICS:
+                                    latest_value = float(latest_value) / 100.0
+                                if resource_type == 'cluster':
+                                    if 'vsan.cluster.{}'.format(given_metric.metricId.label) in VSAN_METRICS:
+                                        self.gauge(
+                                            'vsphere.vsan.cluster.{}'.format(given_metric.metricId.label),
+                                            # for now we only collect the latest value
+                                            float(latest_value),
+                                            tags=[
+                                                'vsphere_cluster:{}'.format(given_metric.metricId.dynamicProperty[0][1])
+                                            ]
+                                            + self._config.base_tags,
+                                            raw=True,
+                                            hostname=self._hostname,
+                                        )
+                                    else:
+                                        self.log.debug(
+                                            "Skipping metric %s because it is not in the list of metrics to collect",
+                                            given_metric.metricId.label,
+                                        )
+                                elif resource_type == 'host':
+                                    if 'vsan.host.{}'.format(given_metric.metricId.label) in VSAN_METRICS:
+                                        self.gauge(
+                                            'vsphere.vsan.host.{}'.format(given_metric.metricId.label),
+                                            # for now we only collect the latest value
+                                            float(latest_value),
+                                            tags=[
+                                                'vsphere_cluster:{}'.format(given_metric.metricId.dynamicProperty[0][1])
+                                            ]
+                                            + ['vsphere_host:{}'.format(given_metric.metricId.dynamicProperty[0][2])]
```

Comment:
these tags shouldn't be needed since we already add these tags to the host

Reply:
I ran the check without adding these tags, and the metrics don't seem to have these tags (only `vcenter_server`, and not `vcenter_cluster` or `vcenter_host`). The vsan check test also doesn't detect the tags, so it fails without adding the tags.

Reply:
These tags won't show up on the metrics themselves like in the output of `datadog-agent check vsphere`, since they are attached to the host and not the metrics. You can either verify they are there by going to the host in the infrastructure list in the backend and viewing the tags on the host, or asserting them in tests using `assert_external_tags`